typing/typecore.ml

```diff
@@ -5216,22 +5264,22 @@ and type_function
   | { pparam_desc = Pparam_val (arg_label, default_arg, pat); pparam_loc }
       :: rest
     ->
-      let ty_arg, ty_res =
+      let ty_arg, ty_res, ty_arg_mono =
```

Comment:
Note this is the reverse of the usual convention that `ty_arg_mono` is the result of `newmono ty_arg`.